src/Microsoft.DocAsCode.MarkdigEngine.Extensions/TripleColon/ChromelessFormExtension.cs

```diff
@@ -68,11 +70,15 @@ public bool TryProcessAttributes(IDictionary<string, string> attributes, out Htm
             htmlAttributes.AddProperty("data-action", action);
             htmlAttributes.AddClass("chromeless-form");
 
+            submitTextArray.Enqueue(submitText);
+
             RenderDelegate = (renderer, obj) =>
             {
+                var s = submitTextArray.Dequeue();
+
                 renderer.Write("<form").WriteAttributes(obj).WriteLine(">");
                 renderer.WriteLine("<div></div>");
-                renderer.WriteLine($"<button disabled=\"disabled\" type=\"submit\">{submitText}</button>");
+                renderer.WriteLine($"<button class=\"button is-primary\" disabled=\"disabled\" type=\"submit\">{s}</button>");
```

Comment:
Why don't you add "submitText" to RenderProperties in obj? So the obj can contain all the information you need for rendering.

Reply:
The problem here is that inside of the "TryProcessAttributes" function, we don't have access to the root RenderProperties object. (We gain access to it in the delegate, but by then it's too late.) This was by design. The only way to hold state here for multiple iterations is at the class level by using multiple variables, or at the class level by using a single collection.

Reply:
I'd like to preserve Jeremy's API, as it seems simple and extensible (I don't want to add methods that are only needed for my extension). However, I think there may be a way to do this by creating a new HTML child block.

Reply:
Sure. That would be great. It will more friendly for others to maintain the code.

Reply:
Thanks @jeahyoun